### ISSUE TITLE
Can not join room with new key

### ISSUE THREAD

**Issue by HenrikSolver** [matrix-org/dendrite#1453]
**Describe the bug**
My server key had a faulty name so I had to create a new one. After configuring dendrite to use the new key as private_key and the old one as old_private_key. I try to join a room I previously been in I get a long list of errors like.
```
ERRO[2020-09-29T19:17:40.552959071Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200929155210-32fc5888d26a/federationtypes.go:422] Check
         Signature validation failed for event "$VyKN3C6hPLL3ZhgJEt2g8uCf1LwoxwvVkfWKMeiZKuw"  error="Bad signature from \"solver.nu\" with ID \"ed25519:i+b2\""
ERRO[2020-09-29T19:17:40.553023874Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200929155210-32fc5888d26a/federationtypes.go:422] Check
         Signature validation failed for event "$imHB8F7XIW90fRaS3Cb42RMHRJHbv1T6lKgcky-YJEs"  error="Bad signature from \"solver.nu\" with ID \"ed25519:i+b2\""
ERRO[2020-09-29T19:17:40.553091081Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200929155210-32fc5888d26a/federationtypes.go:422] Check
         Signature validation failed for event "$1QUoVjDCgDwVOfjImJQZaowikDYwZ5m96wjAqFE2sCU"  error="Bad signature from \"solver.nu\" with ID \"ed25519:i+b2\""
ERRO[2020-09-29T19:17:40.553175525Z] [github.com/matrix-org/gomatrixserverlib@v0.0.0-20200929155210-32fc5888d26a/federationtypes.go:422] Check
         Signature validation failed for event "$Kk4ogJ_6j5AwjWuWbY8S9b_hqS2-9m8bLKD0PozJBVM"  error="Bad signature from \"solver.nu\" with ID \"ed25519:i+b2\""
INFO[2020-09-29T19:17:40.709372636Z] [github.com/matrix-org/dendrite/federationapi/routing/send.go:84] Send
         Received transaction "1600602870541" containing 0 PDUs, 1 EDUs  req.id=YyRsFVAd8elc req.method=PUT req.path=/_matrix/federation/v1/send/1600602870541
ERRO[2020-09-29T19:17:41.066481109Z] [github.com/matrix-org/dendrite/federationsender/internal/perform.go:228] func1
         Failed to process room join response          error="respSendJoin: gomatrixserverlib: event with ID \"$HYAJeMJTK82U_ZGPnJy_N5xLS0PZ9ygpHlNCEdl0Ahg\" is not allowed by its auth events: gomatrixserverlib: event with ID \"$HYAJeMJTK82U_ZGPnJy_N5xLS0PZ9ygpHlNCEdl0Ahg\" is not allowed by its auth_events: eventauth: \"@henrik:solver.nu\" is not allowed to change their membership from \"leave\" to \"leave\"" room_id="!yomrOFwgFXzmeMAbzX:matrix.org" user_id="@henrik:solver.nu"
```
* I can create new rooms
* I can join rooms on other servers I never been in 
* If I disable the name check of the keys and use the old key as private_key, everything works fine. 
* I'm running a server built from beaf4cc00f6f198202742693cef40d71ed0b255e

**Part of config file**
``` 
  # The path to the signing private key file, used to sign requests and events.
  private_key: /local/dendrite/matrix_key.pem
  
  old_private_keys: 
  - private_key: /local/dendrite/matrix_key.pem.old  
    expired_at: 1600979305807
```

**Comment by neilalexander** [comment]
Will look into this tomorrow.

**Comment by neilalexander** [comment]
I suspect the problem is here—I don't think we are handling cases where an event wants to look up expired keys: https://github.com/matrix-org/dendrite/blob/beaf4cc00f6f198202742693cef40d71ed0b255e/serverkeyapi/internal/api.go#L116-L138

I am assuming that your keys don't otherwise appear in `keydb_server_keys` in your database?

**Comment by neilalexander** [comment]
This might actually be a really simple fix—can you try commit `cb2990` from #1454 and see if that fixes the problem?
